swim/join_sender.go

```diff
@@ -112,6 +130,9 @@ func newJoinSender(node *Node, opts *joinOpts) (*joinSender, error) {
 		node: node,
 	}
 
+	// Parse bootstrap hosts into a map
```

Comment:
Possibly, although the fact that it needs to be parsed into a map is an implementation detail of the `joinSender`. This implementation of joinSender does special peer selection based on host and by having it in a map life becomes easier for the joinSender.

Conceptually, it doesn't really matter to the discoverProvider how these nodes are grouped so I don't think it should be a concern of the discoverProvider.